src/transformations/defmodel.hpp

```diff
@@ -0,0 +1,1890 @@
+#if 0
+        return getLonLatOffset(ix, iy, lonOffsetRadian, latOffsetRadian) &&
+               getZOffset(ix, iy, zOffset);
+#endif
+    }
+
+    bool getEastingNorthingZOffset(int /*ix*/, int /*iy*/,
+                                   double & /*eastingOffset*/,
+                                   double & /*northingOffset*/,
+                                   double & /*zOffset*/) const {
+        throw UnimplementedException("getEastingNorthingOffset unimplemented");
+#if 0
+        return getEastingNorthingOffset(ix, iy, eastingOffset,
+                                        northingOffset) &&
+               getZOffset(ix, iy, zOffset);
+#endif
+    }
+
+#ifdef DEBUG_DEFMODEL
+    std::string name() const {
+        throw UnimplementedException("name() unimplemented");
+    }
+#endif
+};
+
+// ---------------------------------------------------------------------------
+
+/** Concept for a GridSet used by EvaluatorIface. Intended to be implemented
+ * by user code */
+template <class Grid = GridConcept> struct GridSetConcept {
+    // The return pointer should remain "stable" over time for a given grid
+    // of a GridSet.
+    const Grid *gridAt(double /*x */, double /* y */) {
+        throw UnimplementedException("gridAt unimplemented");
+    }
+};
+
+// ---------------------------------------------------------------------------
+
+/** Concept for a EvaluatorIface used by Evaluator. Intended to be implemented
+ * by user code */
+template <class Grid = GridConcept, class GridSet = GridSetConcept<>>
+struct EvaluatorIfaceConcept {
+
+    std::unique_ptr<GridSet> open(const std::string & /* filename*/) {
+        throw UnimplementedException("open unimplemented");
+    }
+
+    void geographicToGeocentric(double /* lam */, double /* phi */,
+                                double /* height*/, double /* a */,
+                                double /* b */, double /*es*/, double & /* X */,
+                                double & /* Y */, double & /* Z */) {
+        throw UnimplementedException("geographicToGeocentric unimplemented");
+    }
+
+    void geocentricToGeographic(double /* X */, double /* Y */, double /* Z */,
+                                double /* a */, double /* b */, double /*es*/,
+                                double & /* lam */, double & /* phi */,
+                                double & /* height*/) {
+        throw UnimplementedException("geocentricToGeographic unimplemented");
+    }
+
+    bool isGeographicCRS(const std::string & /* crsDef */) {
+        throw UnimplementedException("isGeographicCRS unimplemented");
+    }
+
+#ifdef DEBUG_DEFMODEL
+    void log(const std::string & /* msg */) {
+        throw UnimplementedException("log unimplemented");
+    }
+#endif
+};
+
+// ---------------------------------------------------------------------------
+
+constexpr double DEFMODEL_PI = 3.14159265358979323846;
+constexpr double DEG_TO_RAD_CONSTANT = 3.14159265358979323846 / 180.;
```

Comment:
Is there a common place for these kinds of things?

Reply:
we do have PROJ constants, but this file is intended to be PROJ agnostic, and unfortunately M_PI isn't completely standard (on Windows)